src/wp-includes/default-filters.php

```diff
@@ -596,9 +596,11 @@
 add_filter( 'block_editor_settings_all', 'wp_add_editor_classic_theme_styles' );
 
 // Global styles can be enqueued in both the header and the footer. See https://core.trac.wordpress.org/ticket/53494.
-add_action( 'wp_enqueue_scripts', 'wp_enqueue_global_styles' );
+add_action( 'init', 'wp_enqueue_global_styles', 1 );
 add_action( 'wp_footer', 'wp_enqueue_global_styles', 1 );
 
+add_action( 'wp_enqueue_scripts', 'wp_add_global_styles_for_blocks' );
```

Comment:
Doesn't this adds the block-level global styles twice to the `global-styles-inline-css` stylesheet when there's a single stylesheet for blocks?

Reply:
Oh that's a good point, I'll have to test that scenario better.

Reply:
I just confirmed and there's no duplication of block-level global styles in the single stylesheet scenario. I removed it from where it was called inside `wp_enqueue_global_styles` in the script loader, so it's only getting called here, independently of the stylesheet setting.